### PR TITLE
Unify naming of index access calls in kernel and docs

### PR DESCRIPTION
**Pull request by Mats-SX**
This is an effort to try and be consistent with the naming we have for index access classes and methods. The selected name is now to use `[unique]indexSeek` for standard index access and `indexSeekByPrefix` for prefix searches.

### CODE REVIEW

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_3/GeneratedQueryStructure.scala [hunk 1]
```diff
@@ -772,8 +772,8 @@ private object Methods {
   val relationshipTypeGetForName = method[ReadOperations, Int]("relationshipTypeGetForName", typeRef[String])
   val nodesGetAll = method[ReadOperations, PrimitiveLongIterator]("nodesGetAll")
   val nodeGetProperty = method[ReadOperations, Object]("nodeGetProperty")
-  val nodesGetFromIndexLookup = method[ReadOperations, PrimitiveLongIterator]("nodesGetFromIndexLookup", typeRef[IndexDescriptor], typeRef[Object])
-  val nodeGetUniqueFromIndexLookup = method[ReadOperations, Long]("nodeGetUniqueFromIndexLookup", typeRef[IndexDescriptor], typeRef[Object])
+  val nodesGetFromIndexLookup = method[ReadOperations, PrimitiveLongIterator]("nodesGetFromIndexSeek", typeRef[IndexDescriptor], typeRef[Object])
```
Comment (Mats-SX) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_3/GeneratedQueryStructure.scala:775]: Will these changes break some kind of backwards compatibility? @systay @boggle

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_2/TransactionBoundQueryContext.scala [hunk 2]
```diff
@@ -129,13 +129,13 @@ final class TransactionBoundQueryContext(graph: GraphDatabaseAPI,
   }
 
   def exactIndexSearch(index: IndexDescriptor, value: Any) =
-    mapToScala(statement.readOperations().nodesGetFromIndexLookup(index, value))(nodeOps.getById)
+    mapToScala(statement.readOperations().nodesGetFromIndexSeek(index, value))(nodeOps.getById)
```
Comment (Mats-SX) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_2/TransactionBoundQueryContext.scala:132]: Will these changes break some kind of backwards compatibility?

Comment (boggle) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_2/TransactionBoundQueryContext.scala:132]: Kernel API is internal so it shouldn't

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_3/GeneratedQueryStructure.scala (hunk 1 above)
Comment (boggle) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_3/GeneratedQueryStructure.scala:775]: This is internal so it shouldn't

**Pull request merged**